src/solver/solverUtils.F90

```diff
@@ -1348,60 +1430,387 @@ subroutine slipVelocitiesFineLevel_block(useOldCoor, t, sps)
           ! Loop over the quadrilateral faces of the viscous
           ! subface.
 
-          do j=BCData(mm)%jcBeg, BCData(mm)%jcEnd
-             do i=BCData(mm)%icBeg, BCData(mm)%icEnd
-
-                ! Compute the coordinates of the centroid of the face.
-                ! Normally this would be an average of i-1 and i, but
-                ! due to the usage of the pointer xFace and the fact
-                ! that x starts at index 0 this is shifted 1 index.
-
-                xc(1) = fourth*(xFace(i+1,j+1,1) + xFace(i+1,j,1) &
-                     +         xFace(i,  j+1,1) + xFace(i,  j,1))
-                xc(2) = fourth*(xFace(i+1,j+1,2) + xFace(i+1,j,2) &
-                     +         xFace(i,  j+1,2) + xFace(i,  j,2))
-                xc(3) = fourth*(xFace(i+1,j+1,3) + xFace(i+1,j,3) &
-                     +         xFace(i,  j+1,3) + xFace(i,  j,3))
-
-                ! Determine the coordinates relative to the center
-                ! of rotation.
-
-                xxc(1) = xc(1) - rotCenter(1)
-                xxc(2) = xc(2) - rotCenter(2)
-                xxc(3) = xc(3) - rotCenter(3)
+          ! The new procedure is less elegant as the previous one.
```

Comment:
Another big loop to check, with a lot of conditionals

Reply:
can we include tests that check for every case (i,j,k/min,max)? this looks very error prone...

Reply:
Are you talking about specific unit tests for this routine? I am onboard but don't really know where to start tbh